com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/app/discovery/ZigBeeNodeServiceDiscoverer.java

```diff
@@ -203,16 +203,12 @@ private void startDiscovery(Set<NodeDiscoveryTask> newTasks) {
         synchronized (discoveryTasks) {
             logger.debug("{}: Node SVC Discovery: starting new tasks {}", node.getIeeeAddress(), newTasks);
 
-            // Remove any tasks that we know are not supported by this device
-            if ((!supportsManagementLqi || node.getNodeDescriptor() != null
-                    && node.getNodeDescriptor().getLogicalType() == LogicalType.UNKNOWN)
-                    && newTasks.contains(NodeDiscoveryTask.NEIGHBORS)) {
+            // Remove router/coordinator-only tasks if the device is possibly an end node.
+            final boolean isPossibleEndDevice = isPossibleEndDevice();
+            if ((!supportsManagementLqi || isPossibleEndDevice)) {
                 newTasks.remove(NodeDiscoveryTask.NEIGHBORS);
             }
-            if ((!supportsManagementRouting || node.getNodeDescriptor() != null
-                    && (node.getNodeDescriptor().getLogicalType() == LogicalType.UNKNOWN
-                            || node.getNodeDescriptor().getLogicalType() == LogicalType.END_DEVICE))
-                    && newTasks.contains(NodeDiscoveryTask.ROUTES)) {
+            if ((!supportsManagementRouting || isPossibleEndDevice)) {
```

Comment:
By changing the logic around here it should not be needed to add the additional ```if supported``` checks in the tasks themselves.